games-action/polymc/polymc-9999.ebuild

```diff
@@ -39,30 +39,41 @@ LICENSE="Apache-2.0 BSD BSD-2 GPL-2+ GPL-3 ISC LGPL-2.1+ LGPL-3+ MIT"
 
 SLOT="0"
 
-IUSE="debug lto test"
+IUSE="debug lto qt6 test"
 REQUIRED_USE="
 	lto? ( !debug )
 "
 
 RESTRICT="!test? ( test )"
 
-MIN_QT="5.12.0"
-QT_SLOT=5
+MIN_QT_5_VERSION="5.12.0"
+MIN_QT_6_VERSION="6.0.0"
 
 QT_DEPS="
-	>=dev-qt/qtconcurrent-${MIN_QT}:${QT_SLOT}
-	>=dev-qt/qtcore-${MIN_QT}:${QT_SLOT}
-	>=dev-qt/qtgui-${MIN_QT}:${QT_SLOT}
-	>=dev-qt/qtnetwork-${MIN_QT}:${QT_SLOT}
-	>=dev-qt/qttest-${MIN_QT}:${QT_SLOT}
-	>=dev-qt/qtwidgets-${MIN_QT}:${QT_SLOT}
-	>=dev-qt/qtxml-${MIN_QT}:${QT_SLOT}
+	!qt6? (
+		>=dev-qt/qtconcurrent-${MIN_QT_5_VERSION}:5
+		>=dev-qt/qtcore-${MIN_QT_5_VERSION}:5
+		>=dev-qt/qtgui-${MIN_QT_5_VERSION}:5
+		>=dev-qt/qtnetwork-${MIN_QT_5_VERSION}:5
+		>=dev-qt/qttest-${MIN_QT_5_VERSION}:5
+		>=dev-qt/qtwidgets-${MIN_QT_5_VERSION}:5
+		>=dev-qt/qtxml-${MIN_QT_5_VERSION}:5
+	)
+
+	qt6? (
+		>=dev-qt/qtbase-${MIN_QT_6_VERSION}:6[concurrent,gui,network,widgets]
+		>=dev-qt/qtsvg-${MIN_QT_6_VERSION}:6
```

Comment:
Is this correct? We don't seem to depend on qtsvg if we use qt5.

Reply:
I guess you could say it's an optional dependency of sorts. Without qtsvg in the Qt6 build, SVG icons won't be loaded, including instance icons and mod / modpack icons on the third-party API explorers. ~Not sure why Qt5 runs fine (or maybe it doesn't, and we should have an extra dependency in there? I wasn't able to test that just yet)~ found another needed dependency :D.

So, the application does run, but it lacks some stuff. I could also put it only on RDEPEND, though that would separate it from the other Qt depends.

Reply:
> So, the application does run, but it lacks some stuff. I could also put it only on RDEPEND, though that would separate it from the other Qt depends.

It should probably be DEPEND because I suspect that if it is not present at build time the icons won't work even if you install qtsvg afterwards.

Reply:
I just tested it with both the qt5 and qt6 builds, it works just fine if you install it afterwards! It probably uses the Qt plugin system to load it at run-time.